src/library/masq.js

```diff
@@ -43,165 +26,134 @@ const dbReady = db =>
+
+    // Create a DB for this profile
+    const db = openOrCreateDB(id)
+    await dbReady(db)
+    await db.putAsync('/', profile)
 
-    await promiseHyperdb.batch(this.dbs.core, batch)
-    await promiseHyperdb.batch(this.dbs.profiles, batch)
+    this._setProfileToLocalStorage(profile)
   }
 
   /**
-   * Get private profiles from core db
+   * Get public profiles from localstorage (id, username, and )
    */
   async getProfiles () {
-    const node = await promiseHyperdb.get(this.dbs.core, '/profiles')
-    if (!node) return []
-
-    const ids = node.value
-    const profilePromises = ids.map(
-      id => promiseHyperdb.get(this.dbs.core, `/profiles/${id}`)
-    )
-    const profileNodes = await Promise.all(profilePromises)
-    const profiles = profileNodes.map(n => n.value)
-    return profiles
+    return this._getProfilesFromLocalStorage()
   }
 
   /**
    * Update an existing profile
    * @param {object} profile The updated profile
    */
   async updateProfile (profile) {
+    // TODO: Check profile
+    this._checkProfile()
     const id = profile.id
     if (!id) throw Error('Missing id')
-
-    await promiseHyperdb.put(this.dbs.core, `/profile/${id}`, profile)
-    await promiseHyperdb.put(this.dbs.profiles, `/profiles/${id}`, profile)
+    await this.profileDB.putAsync('/', profile)
+    this._setProfileToLocalStorage(profile)
   }
 
   /**
    * Add an app to a specified profile
-   * @param {number} profileId The profile id the app belongs to
    * @param {object} app The app
    */
-  addApp (profileId, app) {
-    return this._createResource(profileId, 'apps', app)
+  addApp (app) {
+    return this._createResource('apps', app)
   }
 
   /**
    * Add a device to a specified profile
-   * @param {number} profileId The profile id the app belongs to
    * @param {object} device The device
    */
-  addDevice (profileId, device) {
-    return this._createResource(profileId, 'devices', device)
+  addDevice (device) {
+    return this._createResource('devices', device)
   }
 
   /**
-   * Get all apps attached to a profile id
-   * @param {number} profileId The profile id for which we get the apps
+   * Get all apps of the current profile
    */
-  getApps (profileId) {
-    return this._getResources(profileId, 'apps')
+  getApps () {
+    this._checkProfile()
+    return this._getResources('apps')
   }
 
   /**
-   * Get all devices attached to a profile id
-   * @param {number} profileId The profile id for which we get the devices
+   * Get all devices of the current profile
    */
-  getDevices (profileId) {
-    return this._getResources(profileId, 'devices')
+  getDevices () {
+    this._checkProfile()
+    return this._getResources('devices')
   }
 
   /**
    * Update an app
-   * @param {number} profileId The profile id to which the app is attached
    * @param {object} app The updated app
    */
-  updateApp (profileId, app) {
-    return this._updateResource(profileId, 'apps', app)
+  updateApp (app) {
```

Comment:
Just an idea, 
If we receive only {name: 'newNme} in the parameter, we can prevent the overwrite of current app by:
1) checking the received object properties, reject if we do not have at least 4 properties : id, name, description and image (just an example)
2) can we take the previous version and use the spread operator  like : {...currentApp, app}

Reply:
We have full control over the lib (these methods are called inside our app), so checking should not be as neccessary as for a lib exposed to a user.
For now we check the presence of id in `_updateResource`, that's the only check made.

I'm afraid adding more checks could slow us down, because we may still need to change how a device or app is defined. Maybe we should check the existence of this resource id before updating it, but it should not happen if we control what we do. What do you think @sujeebant ?

Reply:
If I understand correctly @G-Ray you are suggesting that those checks are not useful as they will never fail because only the proper updates will be proposed to the User through the interface ?

Reply:
ok not needed in internal use.